### PR TITLE
hackwa's first goodie

### PR DESCRIPTION
**Pull request by hackwa**
# Goodie Pull Request Template
**What does your Instant Answer do?**
Returns "hackwa is awesome" when trigger is "duckduckhack hackwa"
**What problem does your Instant Answer solve (Why is it better than organic links)?**
Tell the world about duckduckgo's and mine awesomeness.
**What is the data source for your Instant Answer? (Provide a link if possible)**
None
**Why did you choose this data source?**
I read the tutorial :P
**Are there any other alternative (better) data sources?**
No
**What are some example queries that trigger this Instant Answer?**
duckduckhack GitUserName
**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Me
**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
No
**Which existing Instant Answers will this one supercede/overlap with?**
Other folks who did the tutorial maybe
**Are you having any problems? Do you need our help with anything?**
No
**Where did you hear about DuckDuckHack? (For first time contributors)**
DuckSite
**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
snapshot1

## Checklist
Please place an 'X' where appropriate.

```
- X Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- X Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- X Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- X Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [] Tested cross-browser compatibility (I just have linux ^_^)

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 10

    - Windows 7
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 8
        - [] IE 9
        - [] IE 10

    - Windows XP
        - [] IE 7
        - [] IE 8

    - Mac OSX
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] Safari

    - iOS (iPhone)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - iOS (iPad)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - Android
        - [] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera

```

### CODE REVIEW

File: lib/DDG/Goodie/IsAwesome/hackwa.pm [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package DDG::Goodie::IsAwesome::hackwa;
+# ABSTRACT: hackwa's first Goodie
+use DDG::Goodie;
+zci answer_type => "is_awesome_hackwa";
+zci is_cached   => 1;
+name "IsAwesome hackwa";
+description "My first Goodie, it lets the world know that hackwa is awesome";
+primary_example_queries "duckduckhack hackwa";
+secondary_example_queries "optional -- demonstrate any additional triggers";
```
Comment (MrChrisW) [inline lib/DDG/Goodie/IsAwesome/hackwa.pm]: This line can be removed, it's part of the template.

File: t/IsAwesome/hackwa.t [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    )],
+    'duckduckhack hackwa' => test_zci('hackwa is awesome and has successfully completed the DuckDuckHack Goodie tutorial!'),
+    'duckduckhack hackwa is awesome' => undef,
+    # At a minimum, be sure to include tests for all:
+    # - primary_example_queries
+    # - secondary_example_queries
+    #'example query' => test_zci('query'),
+    # Try to include some examples of queries on which it might
+    # appear that your answer will trigger, but does not.
+    #'bad example query' => undef,
```
Comment (MrChrisW) [inline t/IsAwesome/hackwa.t]: These comments can be removed.

**Comment by MrChrisW** [comment]
@hackwa Congratulations on your first Goodie :+1:  I've added some code feedback, please update when you're able to! Thanks

**Comment by hackwa** [comment]
done removing the comments :smile:

**Comment by mintsoft** [comment]
Congats @hackwa

**Comment by mintsoft** [comment]
@MrChrisW the lack of newline spacing out might be a contentious issue here. I don't have any problem with it personally, however @moollaza @jagtalon may have other feelings.

**Comment by hackwa** [comment]
thanks @mintsoft  :grinning:

**Comment by dronov** [comment]
thumb up

**Comment by MrChrisW** [comment]
@mintsoft That's something that I noticed however wasn't too concerned about due to the nature of this Goodie, however for Goodies performing complex operations, newline spacing is probably best to help with readability. Less lines = smaller file :laughing: optimization!

**Comment by mintsoft** [comment]
@MrChrisW Yeah I agree, it seems like an unnecessarily nit-picky thing for these is_awesome goodies but definitely something that'd need sorting on a bigger IA

**Comment by moollaza** [comment]
@hackwa congrats! It's live: https://duckduckgo.com/?q=duckduckhack+hackwa&ia=answer

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's & Issues or even submit more Instant Answers!

**Comment by hackwa** [comment]
Hey thanks a lot! Feels good to have my name on a search engine :D. I'll
try to help in any way I can ^_^.

**Pull request merged**